documentation/Tutorials/WPF/Categories1.md

```diff
@@ -0,0 +1,191 @@
+        }
+      CategoryList.Add(newCategory);
+      Clear();
+      }
+
+    public void Clear()
+      {
+      CategoryName= string.Empty;
+      CategoryDescription=string.Empty;
+      SelectedCategory=null;
+      }
+    }
+  }
+```
+
+This is a fairly long code fragment. Now each part of it will be discussed separately. A large part of the ViewModel is dovoted to represent the data you need in the View.
+
+The first thing you can see is that the ViewModel inherits from ``Screen`` which is a kind of conductor in Caliburn.Micro. As the name says, it will allow the use of a single user control or window.
+
+``BindableCollection<CategoryModel> CategoryList``
+CategoryList is a List containing all categories. It is used in the DatGrid we will define in the View.
+
+``BindableCollection<T>`` is a class derived from ``ObservableCollection<T>`` but has some improvements in event processing at the UI thread. Therefore it is preferred to use a BindableCollection above ``ObservableCollection``.
+
+There is one more thing to discuss here:
+
+``private BindableCollection<CategoryModel> _categoryList = new BindableCollection<CategoryModel>();``
+
+You see the ``CategoryList`` is instantiated here. This avoids null value exceptions. It is good to know you can do this while creating the property.
+
+``CategoryModel SelectedCategory`` represents the selected item. In the View you must bind it. In non-MVVM solutions you might use an eventhandler to handle selecting an item from a list. Using Caliburn.Micro, we will not do it, but instead we obtain the value and use it in the ViewModel. This separates it nicely from the UI. For example, you now can create a test case where you pick a value, assign it to ``SelectedCategory`` and then use this to drive a test. No need to have a working UI.
+
+You can edit the data in the DataGrid in the View directly, but here a different technique is used. The idea is that you select an item you want to edit (or create a new one) and then save the changes or discard them. You also can create new values. To support this, we need a copy for each field in the ``CategoryModel``.
+
+```csharp
+public string CategoryName
+      {
+      get => _categoryName; set
+        {
+        _categoryName = value;
+        NotifyOfPropertyChange(() => CategoryName);
+        }
+      }
+```
+
+The example shows a full property is used. In the setter we use an additional line of code:
+
+``NotifyOfPropertyChange(() => CategoryName);``
+
+This is the Caliburn.Micro way of implementing change notifications. Its use is not restricted to using it in a setter. Any place where you need to tell the UI that a property is changed, just call this method.
+
+Alternatively you can use ``NotifyOfPropertyChange("SelectedCategory");`` which may look more familiar and may be a way to help migrating an existing project to Caliburn.Micro. The advantage of the lambda expression is that the compiler will detect typos for you.
+
```

Comment:
Does it not need to be said that the class should derive ``PropertyChangedBase``?

Reply:
Not here yet. I will explain that in a later example. You do not need it here, because we derive from the Screen class.